### ISSUE TITLE
Question: Veil-Catapult WMI with Impacket?

### ISSUE THREAD

**Issue by shellster** [Veil-Framework/Veil-Evasion#107]
Hi,

I was wondering why Veil Catapult is using either win-exe or wmis for WMI calls instead of using Impacket.  Impacket has the functionality to do DCOM WMI type calls and they provide an example in their samples.  Is there a good technical reason, or do you just need someone to submit a pull request.  If it's the latter, let me know and I'll take a crack at it.

**Comment by ChrisTruncer** [comment]
Hey there,

Catapult was setup to use win-exe and wmis just because that was what we picked at the time.  There's no reason why we didn't use Impacket, and it could probably work great.  So no, no technical reason for not having it included.

If you want to add in support for it, that'd be awesome!  Feel free to send a pull request.  But also, just want to give a heads up, we don't really actively support Catapult as much any more.  Reason is because all of that functionality has been added into Veil-Pillage (payload delivery capabilities plus more).  We keep Veil-Catapult listed just in case for some reason someone needs to specifically use it vs. Pillage.  So, it would be more useful, if you were to submit an addition that adds in Impacket compatibility, to extend the Veil-Pillage module.  But if you also wanted to add it into Catapult, we wouldn't stop that either.

Hope that helps some.
